pkg/packagekit/updates.jsx

```diff
@@ -63,54 +63,48 @@ import callTracerScript from 'raw-loader!./callTracer.py';
 const _ = cockpit.gettext;
 
 // "available" heading is built dynamically
-let STATE_HEADINGS = {};
-let PK_STATUS_STRINGS = {};
-let PK_STATUS_LOG_STRINGS = {};
+const STATE_HEADINGS = {};
+const PK_STATUS_STRINGS = {};
+const PK_STATUS_LOG_STRINGS = {};
 const packageSummaries = {};
 
 function init() {
-    STATE_HEADINGS = {
-        loading: _("Loading available updates, please wait..."),
```

Comment:
This can actually be done in the global scope, outside of any function.  I.e., just have `const STATE_HEADINGS = { loading: _("...") };` above.

Reply:
No, it can't -- `_()` aka gettext() is a dynamic function imported from cockpit, and the page needs to set the locale first. I validated that again with

```diff
--- pkg/packagekit/updates.jsx
+++ pkg/packagekit/updates.jsx
@@ -63,13 +63,14 @@ import callTracerScript from 'raw-loader!./callTracer.py';
 const _ = cockpit.gettext;
 
 // "available" heading is built dynamically
-const STATE_HEADINGS = {};
+const STATE_HEADINGS = {
+    loading: _("Loading available updates, please wait..."),
+};
 const PK_STATUS_STRINGS = {};
 const PK_STATUS_LOG_STRINGS = {};
 const packageSummaries = {};
 
 function init() {
-    STATE_HEADINGS.loading = _("Loading available updates, please wait...");
     STATE_HEADINGS.locked = _("Some other program is currently using the package manager, please wait...");
     STATE_HEADINGS.refreshing = _("Refreshing package information");
     STATE_HEADINGS.uptodate = _("System is up to date");
```

and on a German UI this now does not translate the "Loading available updates..." any more, as at the "static" scope the locale is still "C".

Reply:
Ah, damn.

Reply:
Yes, this hit us many times before. If cockpit is not yet loaded and initialized, then `cockpit.language` is not yet set and therefore any of these strings would not be translated.

Reply:
As discussed on IRC, I fixed the HTML to make cockpit.gettext() work in the global scope, and eliminated the init() function. Looks much nicer now.

Reply:
@marusak has some doubts that this might be a race condition. I reverted to the previous (approved) version, and made a "someday" todo item to revisit this.